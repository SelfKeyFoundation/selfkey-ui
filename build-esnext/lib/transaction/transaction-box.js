import * as React from 'react';
import { withStyles, createStyles } from '@material-ui/core';
import EthereumIcon from '../icons/ethereum';
import SelfkeyIcon from '../icons/selfkey';
import { ModalCloseIcon } from '../icons/modal-close';
const styles = createStyles({
    transactionBox: {
        padding: '90px 30px 40px 30px',
        position: 'relative',
        borderRadius: '4px',
        backgroundColor: '#262f39',
        border: 'solid 1px #303c49',
        boxShadow: '0 0 16px 1px #15222e'
    },
    iconWrapper: {
        position: 'absolute',
        top: '-32px',
        left: '50%',
        transform: 'translate(-50%)'
    },
    icon: {
        width: '64px !important',
        height: '64px !important'
    },
    closeDialogIconWrapper: {
        position: 'absolute',
        left: '-100px',
        top: 0,
        cursor: 'pointer'
    },
    custom: {
        padding: '10px',
        display: 'block',
        backgroundColor: '#1e262d',
        border: 'solid 1px #303c49',
        color: '#fff',
        fontSize: '25px',
        fontWeight: 600,
        fontFamily: 'Lato, arial, sans-serif'
    }
});
const renderIcon = (shortName, classes) => {
    switch (shortName) {
        case 'KEY':
            return React.createElement(SelfkeyIcon, { className: classes.icon });
        case 'KI':
            return React.createElement(SelfkeyIcon, { className: classes.icon });
        case 'ETH':
            return React.createElement(EthereumIcon, { className: classes.icon });
        default:
            return React.createElement("div", { className: classes.custom }, shortName);
    }
};
export const TransactionBox = withStyles(styles)(({ classes, children, cryptoCurrency, closeAction }) => (React.createElement("div", { className: classes.transactionBox },
    React.createElement("div", { className: classes.closeDialogIconWrapper, onClick: closeAction },
        React.createElement(ModalCloseIcon, null)),
    React.createElement("div", { className: classes.iconWrapper }, renderIcon(cryptoCurrency, classes)),
    React.createElement("div", null, children))));
export default TransactionBox;
//# sourceMappingURL=transaction-box.js.map