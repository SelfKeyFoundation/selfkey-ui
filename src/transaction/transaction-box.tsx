import * as React from 'react';
import { WithStyles, withStyles, createStyles } from '@material-ui/core';
import EthereumIcon from '../icons/ethereum';
import SelfkeyIcon from '../icons/selfkey';
import { ModalCloseIcon } from '../theme/selfkey-dark-theme';

const styles = createStyles({
    transactionBox: {
        padding: '90px 30px 40px 30px',
        position: 'relative',
        borderRadius: '4px',
        backgroundColor: '#262f39',
        border: 'solid 1px #303c49',
        boxShadow: '0 0 16px 1px #15222e',
    },
    iconWrapper: {
        position: 'absolute',
        top: '-32px',
        left: '50%',
        transform: 'translate(-50%)'
    },
    icon: {
        width: '64px !important',
        height: '64px !important'
    },
    closeDialogIconWrapper: {
        position: 'absolute',
        left: '-100px',
        top: 0,
        cursor: 'pointer'
    },
    custom: {
        padding: '10px',
        display: 'block',
        backgroundColor: '#1e262d',
        border: 'solid 1px #303c49',
        color: '#fff',
        fontSize: '25px',
        fontWeight: 600,
        fontFamily: 'Lato, arial, sans-serif'
    }
});

export type TransactionBoxProps = {
    cryptoCurrency: string,
    closeAction?: ((event: React.MouseEvent<HTMLElement>) => void),
    children: any
};

const renderIcon = (shortName: string, classes: any) => {
    switch (shortName) {
       case 'KEY':
            return (<SelfkeyIcon className={classes.icon}/>);
        case 'KI':
          return (<SelfkeyIcon className={classes.icon}/>);
        case 'ETH':
            return (<EthereumIcon className={classes.icon}/>);
        default:
            return (<div className={classes.custom}>{shortName}</div>)
    }
};

export const TransactionBox = withStyles(styles)(({ classes, children, cryptoCurrency, closeAction }: TransactionBoxProps & WithStyles<typeof styles>) => (
    <div className={classes.transactionBox}>
        <div className={classes.closeDialogIconWrapper} onClick={closeAction}>
            <ModalCloseIcon />
        </div>
        <div className={classes.iconWrapper}>
            {renderIcon(cryptoCurrency, classes)}
        </div>
        <div>
            {children}
        </div>
    </div>
));

export default TransactionBox;
