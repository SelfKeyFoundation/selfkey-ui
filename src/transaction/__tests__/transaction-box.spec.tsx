import * as React from 'react';
import { create } from 'react-test-renderer';
import { TransactionBox } from '../transaction-box';

describe('Transaction Box', () => {
	it('should show Transaction Box with ETH Icon', () => {
		const component = create(<TransactionBox cryptoCurrency="ETH">TEST</TransactionBox>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should show Transaction Box with KEY Icon', () => {
		const component = create(<TransactionBox cryptoCurrency="KEY">TEST</TransactionBox>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should show Transaction Box with KI Icon', () => {
		const component = create(<TransactionBox cryptoCurrency="KI">TEST</TransactionBox>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should show Transaction Box with Custom text', () => {
		const component = create(<TransactionBox cryptoCurrency="IHT">TEST</TransactionBox>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
