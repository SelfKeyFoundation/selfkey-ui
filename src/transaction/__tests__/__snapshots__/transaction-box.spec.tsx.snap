// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction Box should show Transaction Box with Custom text 1`] = `
<div
  className="Component-transactionBox-0-1-1"
>
  <div
    className="Component-closeDialogIconWrapper-0-1-4"
  >
    <div
      style={
        Object {
          "cursor": "pointer",
          "height": "38px",
          "marginLeft": "779px",
          "marginTop": "-20px",
          "position": "absolute",
          "width": "38px",
        }
      }
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1 Component-root-0-1-6 MuiSvgIcon-fontSizeLarge-9"
        focusable="false"
        height="38px"
        role="presentation"
        viewBox="0 0 38 38"
        width="38px"
      >
        <g
          fill="none"
          fillRule="evenodd"
          id="Symbols"
          stroke="none"
          strokeWidth="1"
        >
          <g
            id="Close-button"
            transform="translate(1.000000, 1.000000)"
          >
            <g
              id="icon-close-modal"
            >
              <circle
                cx="18"
                cy="18"
                fill="#262F39"
                id="Oval-2"
                r="18"
              />
              <g
                fill="#23E6FE"
                fillRule="nonzero"
                id="Shape"
                transform="translate(9.000000, 9.000000)"
              >
                <polygon
                  points="1.81534091 0.639204545 0.639204545 1.81534091 7.82386364 9 0.639204545 16.1846591 1.81534091 17.3607955 9 10.1761364 16.1846591 17.3607955 17.3607955 16.1846591 10.1761364 9 17.3607955 1.81534091 16.1846591 0.639204545 9 7.82386364"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </div>
  </div>
  <div
    className="Component-iconWrapper-0-1-2"
  >
    <div
      className="Component-custom-0-1-5"
    >
      IHT
    </div>
  </div>
  <div>
    TEST
  </div>
</div>
`;

exports[`Transaction Box should show Transaction Box with ETH Icon 1`] = `
<div
  className="Component-transactionBox-0-1-1"
>
  <div
    className="Component-closeDialogIconWrapper-0-1-4"
  >
    <div
      style={
        Object {
          "cursor": "pointer",
          "height": "38px",
          "marginLeft": "779px",
          "marginTop": "-20px",
          "position": "absolute",
          "width": "38px",
        }
      }
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1 Component-root-0-1-6 MuiSvgIcon-fontSizeLarge-9"
        focusable="false"
        height="38px"
        role="presentation"
        viewBox="0 0 38 38"
        width="38px"
      >
        <g
          fill="none"
          fillRule="evenodd"
          id="Symbols"
          stroke="none"
          strokeWidth="1"
        >
          <g
            id="Close-button"
            transform="translate(1.000000, 1.000000)"
          >
            <g
              id="icon-close-modal"
            >
              <circle
                cx="18"
                cy="18"
                fill="#262F39"
                id="Oval-2"
                r="18"
              />
              <g
                fill="#23E6FE"
                fillRule="nonzero"
                id="Shape"
                transform="translate(9.000000, 9.000000)"
              >
                <polygon
                  points="1.81534091 0.639204545 0.639204545 1.81534091 7.82386364 9 0.639204545 16.1846591 1.81534091 17.3607955 9 10.1761364 16.1846591 17.3607955 17.3607955 16.1846591 10.1761364 9 17.3607955 1.81534091 16.1846591 0.639204545 9 7.82386364"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </div>
  </div>
  <div
    className="Component-iconWrapper-0-1-2"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-1 Component-root-0-1-7 Component-icon-0-1-3"
      focusable="false"
      role="presentation"
      viewBox="0 0 44 44"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <rect
          fill="#9418DC"
          height="44"
          rx="9"
          width="44"
        />
        <path
          d="M11.37 20.286l10.027-4.1 9.719 4.205L21.382 5.5 11.37 20.286zm.031 4.55l10.012 5.627 10.212-5.627L21.505 38.5 11.401 24.836zm10.043-6.645l10.15 4.385-10.15 5.508L11 22.456l10.444-4.265z"
          fill="#FFF"
          fillRule="nonzero"
        />
      </g>
    </svg>
  </div>
  <div>
    TEST
  </div>
</div>
`;

exports[`Transaction Box should show Transaction Box with KEY Icon 1`] = `
<div
  className="Component-transactionBox-0-1-1"
>
  <div
    className="Component-closeDialogIconWrapper-0-1-4"
  >
    <div
      style={
        Object {
          "cursor": "pointer",
          "height": "38px",
          "marginLeft": "779px",
          "marginTop": "-20px",
          "position": "absolute",
          "width": "38px",
        }
      }
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1 Component-root-0-1-6 MuiSvgIcon-fontSizeLarge-9"
        focusable="false"
        height="38px"
        role="presentation"
        viewBox="0 0 38 38"
        width="38px"
      >
        <g
          fill="none"
          fillRule="evenodd"
          id="Symbols"
          stroke="none"
          strokeWidth="1"
        >
          <g
            id="Close-button"
            transform="translate(1.000000, 1.000000)"
          >
            <g
              id="icon-close-modal"
            >
              <circle
                cx="18"
                cy="18"
                fill="#262F39"
                id="Oval-2"
                r="18"
              />
              <g
                fill="#23E6FE"
                fillRule="nonzero"
                id="Shape"
                transform="translate(9.000000, 9.000000)"
              >
                <polygon
                  points="1.81534091 0.639204545 0.639204545 1.81534091 7.82386364 9 0.639204545 16.1846591 1.81534091 17.3607955 9 10.1761364 16.1846591 17.3607955 17.3607955 16.1846591 10.1761364 9 17.3607955 1.81534091 16.1846591 0.639204545 9 7.82386364"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </div>
  </div>
  <div
    className="Component-iconWrapper-0-1-2"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-1 Component-root-0-1-8 Component-icon-0-1-3"
      focusable="false"
      role="presentation"
      viewBox="0 0 44 44"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <rect
          fill="#2DA1F8"
          height="44"
          rx="9"
          width="44"
        />
        <path
          d="M33.2899445,22.9203128 C31.3240959,20.9415891 28.2827759,20.6709767 26.0176127,22.1008593 L14.3790117,10.384719 C13.8693908,9.87176033 13.0437885,9.87176033 12.5342002,10.384719 L8.38221571,14.5638959 C7.87259476,15.0768546 7.87259476,15.9078641 8.38221571,16.4207899 C8.89183732,16.9337157 9.71743957,16.9337486 10.2270279,16.4207899 L13.4554644,13.171209 L15.7614951,15.4923429 L12.5330586,18.7419238 C12.0234377,19.2548825 12.0234377,20.0858919 12.5330586,20.5988178 C13.0426796,21.1117436 13.8682818,21.1117765 14.3778702,20.5988178 L17.6063067,17.3492369 L24.1715292,23.9574578 C22.7509506,26.237555 23.0210741,29.2987939 24.9856507,31.2774191 C27.2674805,33.5741936 31.0068426,33.5741936 33.2886398,31.2774191 C35.5704371,28.9806446 35.5704697,25.2167919 33.2886398,22.9200502 L33.2899445,22.9203128 Z M31.4451329,24.7772068 C32.7293777,26.0698627 32.7293777,28.1267859 31.4451329,29.4196059 C30.1608881,30.7124259 28.1173489,30.7122618 26.832941,29.4196059 C25.5485331,28.12695 25.5486962,26.0700268 26.832941,24.7772068 C28.1171858,23.4843868 30.1607251,23.4845509 31.4451329,24.7772068 Z"
          fill="#FFFFFF"
          id="Page-1-Copy"
          transform="translate(21.500000, 21.500000) scale(-1, 1) rotate(45.000000) translate(-21.500000, -21.500000) "
        />
      </g>
    </svg>
  </div>
  <div>
    TEST
  </div>
</div>
`;

exports[`Transaction Box should show Transaction Box with KI Icon 1`] = `
<div
  className="Component-transactionBox-0-1-1"
>
  <div
    className="Component-closeDialogIconWrapper-0-1-4"
  >
    <div
      style={
        Object {
          "cursor": "pointer",
          "height": "38px",
          "marginLeft": "779px",
          "marginTop": "-20px",
          "position": "absolute",
          "width": "38px",
        }
      }
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1 Component-root-0-1-6 MuiSvgIcon-fontSizeLarge-9"
        focusable="false"
        height="38px"
        role="presentation"
        viewBox="0 0 38 38"
        width="38px"
      >
        <g
          fill="none"
          fillRule="evenodd"
          id="Symbols"
          stroke="none"
          strokeWidth="1"
        >
          <g
            id="Close-button"
            transform="translate(1.000000, 1.000000)"
          >
            <g
              id="icon-close-modal"
            >
              <circle
                cx="18"
                cy="18"
                fill="#262F39"
                id="Oval-2"
                r="18"
              />
              <g
                fill="#23E6FE"
                fillRule="nonzero"
                id="Shape"
                transform="translate(9.000000, 9.000000)"
              >
                <polygon
                  points="1.81534091 0.639204545 0.639204545 1.81534091 7.82386364 9 0.639204545 16.1846591 1.81534091 17.3607955 9 10.1761364 16.1846591 17.3607955 17.3607955 16.1846591 10.1761364 9 17.3607955 1.81534091 16.1846591 0.639204545 9 7.82386364"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </div>
  </div>
  <div
    className="Component-iconWrapper-0-1-2"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-1 Component-root-0-1-8 Component-icon-0-1-3"
      focusable="false"
      role="presentation"
      viewBox="0 0 44 44"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <rect
          fill="#2DA1F8"
          height="44"
          rx="9"
          width="44"
        />
        <path
          d="M33.2899445,22.9203128 C31.3240959,20.9415891 28.2827759,20.6709767 26.0176127,22.1008593 L14.3790117,10.384719 C13.8693908,9.87176033 13.0437885,9.87176033 12.5342002,10.384719 L8.38221571,14.5638959 C7.87259476,15.0768546 7.87259476,15.9078641 8.38221571,16.4207899 C8.89183732,16.9337157 9.71743957,16.9337486 10.2270279,16.4207899 L13.4554644,13.171209 L15.7614951,15.4923429 L12.5330586,18.7419238 C12.0234377,19.2548825 12.0234377,20.0858919 12.5330586,20.5988178 C13.0426796,21.1117436 13.8682818,21.1117765 14.3778702,20.5988178 L17.6063067,17.3492369 L24.1715292,23.9574578 C22.7509506,26.237555 23.0210741,29.2987939 24.9856507,31.2774191 C27.2674805,33.5741936 31.0068426,33.5741936 33.2886398,31.2774191 C35.5704371,28.9806446 35.5704697,25.2167919 33.2886398,22.9200502 L33.2899445,22.9203128 Z M31.4451329,24.7772068 C32.7293777,26.0698627 32.7293777,28.1267859 31.4451329,29.4196059 C30.1608881,30.7124259 28.1173489,30.7122618 26.832941,29.4196059 C25.5485331,28.12695 25.5486962,26.0700268 26.832941,24.7772068 C28.1171858,23.4843868 30.1607251,23.4845509 31.4451329,24.7772068 Z"
          fill="#FFFFFF"
          id="Page-1-Copy"
          transform="translate(21.500000, 21.500000) scale(-1, 1) rotate(45.000000) translate(-21.500000, -21.500000) "
        />
      </g>
    </svg>
  </div>
  <div>
    TEST
  </div>
</div>
`;
